src/bokeh/models/annotations/geometry.py

```diff
@@ -217,34 +262,46 @@ def __init__(self, *args, **kwargs) -> None:
         This property is experimental and may change at any point.
     """)
 
-    inverted = Bool(default=False, help="""
-    Inverts the geometry of the box, i.e. applies fill and hatch visuals
-    to the outside of the box instead of the inside. Visuals are applied
-    between the box and its parent, e.g. the frame.
-    """)
+    use_handles = Bool(default=False, help="""
+    Whether to show interaction (move, resize, etc.) handles.
 
-    line_props = Include(ScalarLineProps, help="""
-    The {prop} values for the box.
-    """)
+    If handles aren't used, then the whole annotation, its borders and corners
+    act as if they were interaction handles.
 
-    fill_props = Include(ScalarFillProps, help="""
-    The {prop} values for the box.
+    .. note::
+        This property is experimental and may change at any point.
     """)
 
-    hatch_props = Include(ScalarHatchProps, help="""
-    The {prop} values for the box.
-    """)
+    handles = Struct(
+        all          = Instance(AreaVisuals),           # move, resize
 
-    hover_line_props = Include(ScalarLineProps, prefix="hover", help="""
-    The {prop} values for the box when hovering over.
-    """)
+        move         = Optional(Instance(AreaVisuals)),
+        resize       = Optional(Instance(AreaVisuals)), # sides, corners
 
-    hover_fill_props = Include(ScalarFillProps, prefix="hover", help="""
-    The {prop} values for the box when hovering over.
+        sides        = Optional(Instance(AreaVisuals)), # left, right, top, bottom
+        corners      = Optional(Instance(AreaVisuals)), # top_left, top_right, bottom_left, bottom_right
+
+        left         = Optional(Instance(AreaVisuals)),
+        right        = Optional(Instance(AreaVisuals)),
+        top          = Optional(Instance(AreaVisuals)),
+        bottom       = Optional(Instance(AreaVisuals)),
+
+        top_left     = Optional(Instance(AreaVisuals)),
+        top_right    = Optional(Instance(AreaVisuals)),
+        bottom_left  = Optional(Instance(AreaVisuals)),
+        bottom_right = Optional(Instance(AreaVisuals)),
```

Comment:
More specific "nodes" take precedence over more generic ones. E.g. setting `all` and `corners` and `top_left` with different styles, will result in `top_left` having its own style, all other corners having their own styles (from `corners`) and finally all other nodes will have different styles again (from `all`), There is no immediate mechanism for inheriting properties from more generic nodes, because this is supposed to allow different shapes, etc. in the future. One way for resolving this, which I use in examples and tests, is to clone a more generic nodes and override required properties.

Reply:
I will update the docstring with the explanation.

Reply:
What's the advantage of using a struct variable defining a real model? This just reminds me of that old datetime formatters dict which we eventually changed to a model due to how clunky usage was.

Reply:
Probably none this days. Perhaps one is that I don't have to name it, but I already give this a name in bokehjs.

Reply:
I'd strongly suggest considering switching to a real model. It's much better for UX and documentation IMO. (for that matter, I'm not actually even sure how `Struct` will render in the ref docs, it might just be `Struct` with no additional info)

Reply:
I changed the struct to a proper model named `BoxInteractionHandles`. `Box` suffix is in anticipation that there will be other kinds of handles for other classes of annotations, etc.